08.ls_object/lib/ls_long_format.rb

```diff
@@ -0,0 +1,41 @@
+# frozen_string_literal: true
+
+require_relative './file_information'
+
+class LsLongFormat
+  class << self
+    def output(file_names, directory_path)
+      file_infos = to_file_infos(file_names, directory_path)
+
+      output = []
+
+      output << "total #{file_infos.sum(&:blocks)}" if directory_path
+      output << long_format_of(file_infos)
+
+      output
+    end
+
+    private
+
+    def to_file_infos(file_names, directory_path = nil)
+      file_names.map do |file_name|
+        file_path = directory_path ? "#{directory_path}/#{file_name}" : file_name
+
+        FileInformation.new(file_name, File.lstat(file_path), directory_path)
+      end
+    end
+
+    def long_format_of(file_infos)
+      long_formats = file_infos.map(&:long_format)
+      long_formats = long_formats.transpose.map do |infos|
+        max_length = infos.map { |info| info.to_s.length }.max
+
+        infos.map { |info| info.to_s.rjust(max_length) }
+      end.transpose
+
+      long_formats.map do |file_mode, nlink, user_name, group_name, size, formatted_mtime, file_name|
+        "#{file_mode} #{nlink} #{user_name}  #{group_name}  #{size} #{formatted_mtime} #{file_name.strip}"
+      end
```

Comment:
こうも書ける？けれど、わかりにくいように思う

```suggestion
      long_formats.map { |long_format| format("%s %s %s  %s  %s %s %s" % long_format) }
```